mitxgraders/listgrader.py

```diff
@@ -734,3 +741,33 @@ def check_response(self, answer, student_input, **kwargs):
                                         partial_credit=self.config['partial_credit'])
 
         return result
+
+def demand_no_nonempty(obj):
+    """
+    Recursively search through all lists and tuples in obj.
+    Make sure that all strings are non-empty.
+    If we find a dictionary, if it has an 'expect' key that is a string, make sure
+    it is non-empty also.
+    """
+    if isinstance(obj, list) or isinstance(obj, tuple):
+        for item in obj:
+            demand_no_nonempty(item)
+    else:
+        test = obj
+        if isinstance(test, dict):
+            if 'expect' in test:
+                test = test['expect']
+            else:
+                return
+        # We're doing this BEFORE validation, so could be given text or a string
+        if isinstance(test, six.text_type) or isinstance(test, six.string_types):
```

Comment:
You only need `isinstance(test, six.string_types)` here ( `(basestring,)` in 2, `(str,)` in 3)

Reply:
Happy to make the change, but if someone imports unicode literals, then only doing `string_types` will fail though, right?

Reply:
Should work... `basestring` is the superclass of string and unicode.

```python
import six

# Python 2.7
isinstance(u'cat', six.string_types) # True
isinstance('cat', six.string_types) # True
```

Reply:
Good point. Now fixed.